src/Selection.js

```diff
@@ -218,44 +222,46 @@ class Selection {
       if (!collides) return
     }
 
-    let result = this.emit(
-      'beforeSelect',
-      (this._initialEventData = {
-        isTouch: /^touch/.test(e.type),
-        x: pageX,
-        y: pageY,
-        clientX,
-        clientY,
-      })
-    )
+    setTimeout(() => {
```

Comment:
`setTimeout(fn, 0)` is discouraged.
Better use `setImmediate`

Anyway, what's the point of this delay?

Reply:
no no no, this code is used in a browser. Browsers don’t support it 

https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate